Blockzilla/IntroViewController.swift

```diff
@@ -26,6 +26,104 @@ struct IntroViewControllerUX {
     static let FadeDuration = 0.25
 }
 
+protocol PageControlDelegate {
+    func selectedIndex(_ index:Int)
+}
+
+class PageControl: NSObject {
+    
+    var currentPage = 0
+    var numberOfPages = 0
+    var hidesForSinglePage = false
+    
+    var stack : UIStackView? = nil
+    var delegate: PageControlDelegate?
+    
+    init(numberOfPages: Int) {
+        super.init()
+
+        var buttonArray: [UIButton]!
+        
+        // Ensure we have at least one button
+        if numberOfPages == 0 {
+            return
+        }
+        
+        for _ in 0..<numberOfPages {
+            let button = UIButton()
+            button.setImage(drawDot(), for: .normal)
+            buttonArray.append(button)
+        }
+        
+        // Style the StackView
+        stack = UIStackView(arrangedSubviews: buttonArray)
+        stack?.spacing = 20
+        stack?.distribution = .fillEqually
+        stack?.alignment = .center
+        
+        // Enable the buttons to be tapped to switch to next page
+        buttonArray.forEach({ button in
```

Comment:
optional: feel free to use trailing closure syntax (at least on Firefox iOS the devs use it heavily, not sure about Focus convention yet)